gamemodes/terrortown/entities/weapons/weapon_ttt_smokegrenade.lua

```diff
@@ -1,7 +1,97 @@
+---
+-- @class SWEP
+-- @section weapon_ttt_smokegrenade
+
 if SERVER then
 	AddCSLuaFile()
 end
 
+local flags = {FCVAR_NOTIFY, FCVAR_ARCHIVE, FCVAR_REPLICATED}
+---
+-- @realm server
+CreateConVar("ttt_smokegrenade_proj_throw_force", "1", flags)
+
+---
+-- @realm server
+CreateConVar("ttt_smokegrenade_proj_detonate_timer", "5", flags)
+
+---
+-- @realm server
+CreateConVar("ttt_smokegrenade_proj_disarm_hit_players", "1", flags)
+
+---
+-- @realm server
+CreateConVar("ttt_smokegrenade_proj_damage_trigger_behavior", "1", flags)
+
+---
+-- @realm server
+CreateConVar("ttt_smokegrenade_proj_damage_trigger_instant_explosion", "0", flags)
+
+---
+-- @realm server
+CreateConVar("ttt_smokegrenade_proj_damage_trigger_projectile_explosion", "1", flags)
+
+---
+-- @realm server
+CreateConVar("ttt_smokegrenade_proj_prevent_damage_chain_reaction", "1", flags)
+
+---
+-- @realm server
+CreateConVar("ttt_smokegrenade_proj_use_animation_timers", "0", flags)
+
+---
+-- @realm server
+CreateConVar("ttt_smokegrenade_proj_spherical_physics_radius", "0", flags)
```

Comment:
ah right, we said that the `_disarm_hit_players` convars should be merged into a single one so that either all grenades or none disarm players on hit.

Reply:
Mhhh i feel like this would also be something i'd rather move to a per-grenade variable. 
What is the use case to toggle this for all?

Reply:
Personally I made it this way only because the conf grenade makes the most sense to make people drop stuff on impact, making a flashbang do that could be the thing that spells certain death for a player. There are other grenades such as the freeze grenade where a disarming effect might be the best you could hope for since its AOE is small.

For anything that I'd wind up gutting, I'd have to leave hooks in the correct spot to move them to SMORE, so when thinking about that, consider what sort of thing you'd name the hook to go where the code was. Would a client hook fit that deeply nestled into the animation logic of a grenade throw? What would its signature be?